slack/rtm/client.py

```diff
@@ -448,37 +459,27 @@ async def _dispatch_event(self, event, data=None):
                     # close/error callbacks.
                     break
 
-                if inspect.iscoroutinefunction(callback):
+                if self.run_async or inspect.iscoroutinefunction(callback):
                     await callback(
                         rtm_client=self, web_client=self._web_client, data=data
                     )
                 else:
-                    self._execute_in_thread(callback, data)
+                    await self._execute_in_thread(
+                        callback=callback, web_client=self._web_client, data=data
+                    )
             except Exception as err:
                 name = callback.__name__
                 module = callback.__module__
                 msg = f"When calling '#{name}()' in the '{module}' module the following error was raised: {err}"
                 self._logger.error(msg)
                 raise
 
-    def _execute_in_thread(self, callback, data):
+    async def _execute_in_thread(self, callback, web_client, data):
         """Execute the callback in another thread. Wait for and return the results."""
-        web_client = WebClient(
-            token=self.token,
-            base_url=self.base_url,
-            ssl=self.ssl,
-            proxy=self.proxy,
-            headers=self.headers,
+        future = self.thread_pool_executor.submit(
```

Comment:
You're right. We should abandon the executor when destroying `RTMClient`. As we've discussed the possibility to remove a thread pool from this class, I may remove this part entirely. Otherwise, I will change the code to clean up executors when shutting down.

Reply:
I've removed the ThreadPoolExceutor in https://github.com/slackapi/python-slackclient/pull/662/commits/d9aa23802e92fe3f61ff2fc90271af3b941b9e84